tiledb/sm/filesystem/vfs.cc

```diff
@@ -45,6 +45,12 @@
 #include <sstream>
 #include <unordered_map>
 
+#define TILEDB_CHECK_SUPPORTED_FS                                  \
```

Comment:
Let's just make this a function within `VFS`.

Reply:
And then replace all occurences of its usage with `RETURN_NOT_OK(check_supported_fs(uri))`

Reply:
Resolving for now - will make a note to add this separately as we don't currently have a "not initialized" state for all of the backends.